File: TIWPR-PROJEKT-2/backend/chessBase/src/main/java/pl/tiwpr/chessbase/model/Game.java
package pl.tiwpr.chessbase.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "games")
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column()
    private String event;

    @Column
    private LocalDate date;

    @Column()
    private Integer round;

    @Version
    @Column(name="OPTCLOCK", nullable = false, columnDefinition = "integer DEFAULT 1")
    private Long version;

    @ManyToOne
    @JoinColumn(name = "whitePlayer_id", referencedColumnName = "id")
    private Player whitePlayer;

    private Integer whiteELO;


    @ManyToOne
    @JoinColumn(name = "blackPlayer_id", referencedColumnName = "id")
    private Player blackPlayer;

    private Integer blackELO;

    private String pgn;

    @Enumerated(EnumType.STRING)
    private Result result;

}

File: frontend/src/models/game.ts
import { Player } from "./player";

export type CreateGame = {
    id: number | null;
    event: string;
    date: string;
    round: number;
    whitePlayer: number;
    blackPlayer: number;
    whiteELO: number;
    blackELO: number;
    pgn: string;
    result: string;
}

export type ViewGame = {
    id: number;
    event: string;
    date: string;
    round: number;
    whitePlayer: Player;
    blackPlayer: Player;
    whiteELO: number;
    blackELO: number;
    pgn: string;
    result: string;
}